src/middleware/host-limit.ts

```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+
+export function restrictHostnames<TFn extends express.Router>(
+	allowedHostnames: string[],
+	fn: TFn
+): (req: express.Request, res: express.Response, next: () => void) => void | TFn {
+	return (req: express.Request, res: express.Response, next: () => void) => {
+		if (!allowedHostnames.includes(req.hostname)) {
+			return fn(req, res, next);

```

Comment:
Nit, just to keep it consistent with the rest of the codebase

```suggestion
): (request: express.Request, response: express.Response, next: () => void) => void | TFn {
	return (request: express.Request, response: express.Response, next: () => void) => {
		if (!allowedHostnames.includes(request.hostname)) {
			return fn(request, response, next);

```

Also is there a reason to use this rather than something like https://www.npmjs.com/package/vhost, which Express already provides? Though vhost hasn't had any updates in years, and seems to have [issues with reverse proxies](https://github.com/expressjs/vhost/issues/20), so maybe it's not the best idea

Reply:
It looks like vhost doesn't support multiple domains outside of constructing a regex that matches all the domains

In situations like this where the logic is simple, I would rather implement it myself and have full control - especially due to the sort of stuff you mentioned with outdated code and issues

Reply:
Very fair 👍